src/webots/maths/WbVector2.hpp

```diff
@@ -129,6 +130,18 @@ class WbVector2 {
   // returns a unit vector with the same direction: / length
   void normalize() { *this /= length(); }
   WbVector2 normalized() const { return *this / length(); }
+
+  void clamp() {  // clamp to FLT_MAX
+    if (mX > FLT_MAX)
+      mX = FLT_MAX;
+    else if (mX < -FLT_MAX)
+      mX = -FLT_MAX;
+    if (mY > FLT_MAX)
+      mY = FLT_MAX;
+    else if (mY < -FLT_MAX)
+      mY = -FLT_MAX;
+  }
```

Comment:
```suggestion
  void clamp(double min = -FLT_MAX, double max = FLT_MAX) {
    if (mX > max)
      mX = max;
    else if (mX < min)
      mX = min;
    if (mY > max)
      mY = max;
    else if (mY < min)
      mY = min;
  }
```

Since we are introducing a clamp format it doesn't cost much to make it generic.